geth/transactions/txqueue_manager_test.go

```diff
@@ -61,19 +63,53 @@ func (s *TxQueueTestSuite) TearDownTest() {
 	s.client.Close()
 }
 
-func (s *TxQueueTestSuite) setupTransactionPoolAPI(account *common.SelectedExtKey, nonce hexutil.Uint64, gas hexutil.Big, txErr error) {
-	s.txServiceMock.EXPECT().GetTransactionCount(gomock.Any(), account.Address, gethrpc.PendingBlockNumber).Return(&nonce, nil)
-	s.txServiceMock.EXPECT().GasPrice(gomock.Any()).Return(big.NewInt(10), nil)
-	s.txServiceMock.EXPECT().EstimateGas(gomock.Any(), gomock.Any()).Return(&gas, nil)
-	s.txServiceMock.EXPECT().SendRawTransaction(gomock.Any(), gomock.Any()).Return(gethcommon.Hash{}, txErr)
+func (s *TxQueueTestSuite) setupTransactionPoolAPI(tx *common.QueuedTx, config *params.NodeConfig, account *common.SelectedExtKey, nonce *hexutil.Uint64, gas *hexutil.Big, txErr error) {
```

Comment:
what do you think about preparing tx earlier and passing it to this method? it looks like it will cleanup an interface

Reply:
Hmm, what do you mean by "preparing tx earlier and passing it to this method"? Isn't that already how it works?

Reply:
i thought  that it is possible to pass types.Transaction instead of `tx *common.QueuedTx, config *params.NodeConfig, account *common.SelectedExtKey, nonce *hexutil.Uint64, gas *hexutil.Big`

Reply:
I see what you mean, I may be able to further reduce the code using `types.Transaction`. I'll look into it.

Reply:
On a second glance, it will just result in moving lines elsewhere because one has to form both `types.Transaction` and `common.QueuedTx` object in the same case. In other words, if I define a `types.Transaction` object in the test case and pass it to `setupTransactionPoolAPI` instead, that means I can remove forming a new object from `s.rlpEncode` and it doesn't make a difference about how I handle rest of the args such as gas, gas price and nonce.

But it's definitely possible to reduce the amount of args there by having default test values so I'm going to do that instead.

Reply:
ok, thanks for checking